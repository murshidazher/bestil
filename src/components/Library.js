import React from 'react';
import Card from './Card';
import './Library.css';


const Library = ({books}) => {

    if (books === undefined || books.length === 0) 
        return <h1>No Books Found</h1>;
    else {
        return (
            <div className="library">
                 {books.map((book, i) => {
                     
                     let randomPrice = Math.floor(Math.random() * 53) + 18;
                     let discountPrice = Math.floor(Math.random() * randomPrice) + 9;
                     let status = (Math.floor(Math.random() * 2) === 1) ? 'Available': 'Sold Out';
    
                    return <Card 
                            key={book.id} 
                            id={book.id} 
                            cover={
                                (book.volumeInfo.imageLinks !== undefined) ?book.volumeInfo.imageLinks.smallThumbnail.replace('&edge=curl','')
                                : 'https://freefrontend.com/assets/img/html-funny-404-pages/SVG-Animation-404-Page.png'
                            } 
                            name={
                                book.volumeInfo.title
                            } 
                            author={
                                (book.volumeInfo.authors !== undefined) ? 
                                book.volumeInfo.authors 
                                : 'Unknown Author'
                            } 
                            desc={
                                (book.volumeInfo.description !== undefined) ? 
                                book.volumeInfo.authors 
                                : 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Velit nobis labore fuga, enim, nulla vel amet assumenda expedita, consequuntur ipsum delectus? Excepturi, eum! Minima voluptates dicta beatae aliquid, consectetur omnis?'} 
                            status={status} 
                            pbd={randomPrice} 
                            pad={discountPrice}/>
                } )}
            </div>
        );
    }
    
}

export default Library;

