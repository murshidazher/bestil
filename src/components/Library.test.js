import React from 'react';
import { shallow } from 'enzyme';
import Library from './Library';

it('except to render Library component', () => {
    const mockBooks = [{
        id: 1,
        volumeInfo: {
            title: 'Book about John Doe',
            authors: ['Jane Jackson', 'Janet Jackson'],
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Velit nobis labore fuga',
            imageLinks: {
                smallThumbnail:
                    'https://books.google.com/books/content?id=SQMQQyIaACYC&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;source=gbs_api'
            }
        }

    }]
    expect(shallow(<Library books={mockBooks} />)).toMatchSnapshot();
})

it('except to render Library component with undefined imageLinks', () => {
    const mockBooks = [{
        id: 1,
        volumeInfo: {
            title: 'Book about John Doe',
            authors: ['Jane Jackson', 'Janet Jackson'],
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Velit nobis labore fuga',
            imageLinks: {
                smallThumbnail:
                    'https://books.google.com/books/content?id=SQMQQyIaACYC&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;source=gbs_api'
            }
        }

    }]
    expect(shallow(<Library books={mockBooks} />)).toMatchSnapshot();
})

it('except to render Library component with undefined authors', () => {
    const mockBooks = [{
        id: 1,
        volumeInfo: {
            title: 'Book about John Doe',
            authors: undefined,
            description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Velit nobis labore fuga',
            imageLinks: {
                smallThumbnail:
                    'https://books.google.com/books/content?id=SQMQQyIaACYC&amp;printsec=frontcover&amp;img=1&amp;zoom=5&amp;source=gbs_api'
            }
        }

    }]
    expect(shallow(<Library books={mockBooks} />)).toMatchSnapshot();
})

it('except to render Library component with undefined description', () => {
    const mockBooks = [{
        id: 1,
        volumeInfo: {
            title: 'Book about John Doe',
            authors: ['Jane Jackson', 'Janet Jackson'],
            description: undefined,
            imageLinks: undefined
        }

    }]
    expect(shallow(<Library books={mockBooks} />)).toMatchSnapshot();
})

it('except to return no books found', () => {
    let mockBooks = []
    expect(shallow(<Library books={mockBooks} />)).toEqual({});

    mockBooks = undefined;
    expect(shallow(<Library books={mockBooks} />)).toEqual({});
})

