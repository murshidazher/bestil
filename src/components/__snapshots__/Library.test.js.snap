// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`except to render Library component 1`] = `ShallowWrapper {}`;

exports[`except to render Library component with undefined authors 1`] = `ShallowWrapper {}`;

exports[`except to render Library component with undefined description 1`] = `ShallowWrapper {}`;

exports[`except to render Library component with undefined imageLinks 1`] = `ShallowWrapper {}`;
